### ISSUE TITLE
shakeThreads=0 doesn't seem to work on OSX

### ISSUE THREAD

**Issue by afranchuk** [ndmitchell/shake#657]
Specifically on Mac OSX Mojave, setting shakeThreads to 0 runs with 1 thread.

**Comment by ndmitchell** [comment]
Did you compile your Haskell program with `-threaded` or not?

Here's the code that computes the number of threads if you pick 0. Can you suggest how to adjust it? https://github.com/ndmitchell/shake/blob/a85aa20fb2a134453ad90b1875e754978c494699/src/General/Extra.hs#L113-L126

**Comment by afranchuk** [comment]
Unfortunately I don't have a mac myself. My program is built the same (using stack) on windows, mac, and linux, and someone using it on mac said it was only using one core (and I saw the behavior to verify). Manually specifying `-jn` (where `n` > 0) worked as expected, so I think that indicates stack is including `-threaded` when building.

I will try to get onto a mac system to see what these values are. It's possible that `getNumProcessors` is where the issue lies...

**Comment by ndmitchell** [comment]
It definitely works on some Mac's, and I test that in the CI, so it's not as simple as "a Mac". I have no idea what else might be a relevant variable though.

**Comment by afranchuk** [comment]
Yeah I believe that. I'll try to get an equivalent system to test on, or maybe ask my coworker to try it out for me.
